microcosm_flask/audit.py

```diff
@@ -45,6 +46,33 @@ def should_skip_logging(func):
     return getattr(func, SKIP_LOGGING, False)
 
 
+@contextmanager
```

Comment:
This is the new logic. It's simple and powerful.

My main question is: should we use `X-Request-Debug` as the header name so that debugging gets passed through to downstream requests?

 - Advantage: it's useful
 - Disadvantage: passing specific log component names is often useless across different components

One possible solution is to simplify to a boolean value and always set the root logger only.